helm/alfresco-content-services/charts/alfresco-content-common/templates/_helpers-activemq.tpl

```diff
@@ -7,24 +7,13 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- end -}}
 
 {{- define "activemq.config" -}}
-{{- if .Values.activemq.enabled -}}
-  ACTIVEMQ_URL: {{ printf "nio://%s-broker:61616" (include "activemq.fullname" .) }}
-  ACTIVEMQ_USER: {{ .Values.activemq.adminUser.username | quote | default "" }}
-  ACTIVEMQ_PASSWORD: {{ .Values.activemq.adminUser.password |  quote | default "" }}
-{{- else -}}
-  ACTIVEMQ_URL: {{ .Values.messageBroker.url | quote }}
-  ACTIVEMQ_USER: {{ .Values.messageBroker.user | quote | default ""  }}
-  ACTIVEMQ_PASSWORD: {{ .Values.messageBroker.password | quote | default "" }}
-{{- end -}}
+  ACTIVEMQ_URL: "$BROKER_URL"
+  ACTIVEMQ_USER: "$BROKER_USERNAME"
+  ACTIVEMQ_PASSWORD: "$BROKER_PASSWORD"
```

Comment:
I've done some manual testing and this is not supported by kubernetes as it is.

Looking at https://kubernetes.io/docs/tasks/inject-data-application/define-interdependent-environment-variables/#define-an-environment-dependent-variable-for-a-container it looks like that to properly reference another variabile value is to define the helper block like:

```helm
{{- define "activemq.config" -}}
- name: ACTIVEMQ_URL
  value: $(BROKER_URL)
- name: ACTIVEMQ_USER
  value: $(BROKER_USERNAME)
- name: ACTIVEMQ_PASSWORD
  value: $(BROKER_PASSWORD)
{{- end -}}
```

and then reference that within `spec.containers.env` of deployments like:

```helm
          envFrom:
          - configMapRef:
              # config map to use, defined in config-transform-router.yaml
              name: {{ template "alfresco.shortname" . }}-router-configmap
          - secretRef:
              name: {{ default (printf "%s-brokersecret" (include "content-services.shortname" .)) .Values.messageBroker.existingSecretName }}
          env:
            {{- include "activemq.config" . | nindent 12 }}
```

and of course remove `activemq.config` from the configmap template that become unnecessary.

Would you mind trying with this approach?

Reply:
@gionn sorry about that, should have remembered that env var trick doesn't work via configmaps

was thinking of making similar changes for the other secrets in the chart....would you prefer those in their own MRs or lumped within this one?

Reply:
in the long run we want to use this approach for every secrets, but for this PR just focus on broker credentials so we can merge earlier

Reply:
ok, once this is approved i'll create separate MRs for the other ones I come across as we roll it out for our purposes, thanks

Reply:
@gionn saw the pipeline was failing on the `sync-service` not being able to connect to ActiveMQ

pretty sure this is because the `sync-service` configmap uses the secret env variables to pass into `JAVA_OPTS` and so isn't expanding them...got a thought about how to fix this that i will push momentarilly

will also rename "activemq.config" to "activemq.env" to make it clearer what it's used for